### PR TITLE
New build system, should work with libpack.

### PR DESCRIPTION
**Pull request by erikh**
@LK4D4 here you go, should be ready to use

Others: we decided to bang out a simple build system for the time being, and aim to have it go-gettable later.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,19 @@
+FROM golang:1.4
+
+RUN apt-get update && apt-get install build-essential pkg-config cmake -y
+
+ENV PKG /go/src/github.com/docker/docker-network
+
+# Get libgit2
+ENV LIBGIT2=github.com/tiborvass/git2go
+ENV LIBGIT2_ORIG=github.com/libgit2/git2go
+RUN go get -d ${LIBGIT2} && \
```
Comment (LK4D4) [inline Dockerfile:10]: can't we just git clone this repo to `github.com/libgit2/git2go` to avoid moving code around?

Comment (LK4D4) [inline Dockerfile:10]: Also we can clone `go_backends` right here with `-b`. We can even reduce time of clone with `--single-branch`

Comment (erikh) [inline Dockerfile:10]: not everything has to be super optimized. :) however, I will do this.

**Comment by erikh** [comment]
PTAL

**Comment by LK4D4** [comment]
@erikh What exactly changed?

**Pull request merged**